nbgitpuller/handlers.py

```diff
@@ -73,8 +132,21 @@ def get(self):
             self.set_header('content-type', 'text/event-stream')
             self.set_header('cache-control', 'no-cache')
 
-            gp = GitPuller(repo, repo_dir, branch=branch, depth=depth, parent=self.settings['nbapp'])
+            # if content_provider is specified then we are dealing with compressed
+            # archive and not a git repo
+            if content_provider is not None:
+                pm = self.setup_plugins(content_provider)
+                req_args = {k: v[0].decode() for k, v in self.request.arguments.items()}
+                download_q = Queue()
+                req_args["progress_func"] = lambda: self._wait_for_sync_progress_queue(download_q)
+                req_args["download_q"] = download_q
+                req_args["repo_parent_dir"] = repo_parent_dir
+                hf_args = {"query_line_args": req_args}
+                results = pm.hook.handle_files(**hf_args)
```

Comment:
```suggestion
                results = plugin_manager.hook.handle_files(query_line_args=req_args)
```

Reply:
It seems a bit weird to me that handle_files would be called directly with "query_line_args", when it in practice gets quite a bit of args explicitly set here.

That makes me think the name query_line_args is a bit misleading. I wonder about the handle_files function signature a bit, should it really just take a single dict with misc key/value's, if so, perhaps we want to make it accept **kwargs which would fit the bill for a misc scrap of args.

Hmmm... I hope to learn more and I don't have a clear suggestion based on my current understanding, but the function signature of handle_files seems a bit off.